src/loki/status_action.cc

```diff
@@ -1,8 +1,49 @@
+#include <boost/property_tree/json_parser.hpp>
+#include <iostream>
+#include <sstream>
+
+#include "baldr/tilehierarchy.h"
+#include "config.h"
 #include "loki/worker.h"
+#include "proto/status.pb.h"
 
 namespace valhalla {
 namespace loki {
-void loki_worker_t::status(Api&) const {
+void loki_worker_t::status(Api& request) const {
+
+  auto* status = request.mutable_status();
+
+  // only return more info if explicitly asked for (can be very expensive)
+  // bail if we wont be getting extra info
+  if (!request.options().verbose() || !allow_verbose)
+    return;
+
+  // get _some_ tile
+  static baldr::graph_tile_ptr tile = nullptr;
+  if (!tile) {
+    for (const auto& tile_id : reader->GetTileSet()) {
+      tile = reader->GetGraphTile(tile_id);
+      if (tile->id().level() < baldr::TileHierarchy::GetTransitLevel().level &&
+          tile->header()->nodecount() > 0) {
+        break;
+      }
+    }
```

Comment:
```suggestion
  for (const auto& tile_id : reader->GetTileSet()) {
    tile = reader->GetGraphTile(tile_id);
    if (tile->id().level() < baldr::TileHierarchy::GetTransitLevel().level &&
        tile->header()->nodecount() > 0) {
      break;
    }
```

Reply:
One question here @kevinkreiser : you want to acquire a tile on _every_ request for the scenario where the tileset is changing, right?

Reply:
then there's no point for a `static tile` anymore I guess

Reply:
oh crap i didnt even see that it was static! yeah we cant use static in this way anyway as its not threadsafe. you can make it threadsafe by using a static initializer though. to do that you can take this loop and throw it into an anonymous namespace above and then initialize the static local variable with that:

```c++
namespace {
  auto get_graphtile_with_nodes_and_edges(valhalla::baldr::GraphReader& reader) {
    graph_tile_ptr tile = nullptr;
    for (const auto& tile_id : reader->GetTileSet()) {
      tile = reader->GetGraphTile(tile_id);
      if (tile->id().level() < baldr::TileHierarchy::GetTransitLevel().level &&
          tile->header()->nodecount() > 0) {
        break;
      }
    }
    return tile;
  }
}

...

namespace valhalla {
namespace loki {
  void loki_worker_t::status(Api& request) const {
    ...
    static tile = get_graphtile_with_nodes_and_edges(reader);  
    ...
  }
}
}
...
```

Reply:
Haha thx good we talked about it then.. and yeah makes sense about thread safety